lib/ontologies_api_client/models/class.rb

```diff
@@ -61,10 +64,27 @@ def self.find(id, ontology, params = {})
 
         def self.search(*args)
           query = args.shift
+
           params = args.shift || {}
+
           params[:q] = query
+
           raise ArgumentError, "You must provide a search query: Class.search(query: 'melanoma')" if query.nil? || !query.is_a?(String)
-          HTTP.post("/search", params)
+
+
+          search_result = federated_get(params) do |url|
+            "#{url}/search"
+          end
+          merged_collections = {results: [], errors: []}
+          search_result.each do |result|
+            if result.collection
+              merged_collections[:results].concat(result.collection)
+            elsif result.errors
+              merged_collections[:errors] << result.errors
```

Comment:
here you need to check if we handle errors the same way in the other PR.

Reply:
the PR https://github.com/ontoportal-lirmm/ontologies_api_ruby_client/pull/17

Reply:
Yes I used the errors I get from the federation helper, here it's different because we get the results in collections this is why I added this if

Reply:
by the way here you should change `results` by `collection`, otherwise you changing the endpoint expected schema.